login/service_whitebox_test.go

```diff
@@ -66,10 +67,26 @@ func TestValidOAuthAccessToken(t *testing.T) {
 		AccessToken: configuration.GetGithubAuthToken(),
 		TokenType:   "Bearer",
 	}
-
 	emails, err := loginService.getUserEmails(context.Background(), accessToken)
-	assert.Nil(t, err)
-	assert.NotEmpty(t, emails)
+	var trials int // Number of tries to reach GitHub
+	if err != nil {
+		for trials = 0; trials < 10; trials++ {
+			emails, err = loginService.getUserEmails(context.Background(), accessToken)
+			if err == nil {
+				assert.Nil(t, err)
+				assert.NotEmpty(t, emails)
+				break
+			}
+			time.Sleep(5 * time.Second) // Pause before the next retry
+		}
+	} else {
+		assert.Nil(t, err)
+		assert.NotEmpty(t, emails)
+	}
+	if trials == 10 {
+		t.Error("Test failed, Maximum Retry limit reached", err) // Test failed after trial for 10 times
```

Comment:
This only holds true when `err != nil`. Otherwise it means we succeeded in the last attempt, right?